File: src/main/java/main/sorting/impl/BubbleSort.java
package main.sorting.impl;

import main.sorting.Sort;

import java.util.Arrays;

public class BubbleSort implements Sort<int[]> {
	private int[] table;

	public int[] getSortedData(int[] tab) {
		this.table = tab;
		sort();
		return table;
	}

	private void sort() {
		for (int i = 0; i < table.length; i++) {
			boolean swapped = false;
			for (int j = 0; j < table.length - 1; j++) {
				if (table[j] > table[j + 1]) {
					swap(j, j + 1);
					swapped = true;
				}
			}
			if (!swapped) break;
		}
	}

	private void swap(int left, int right) {
		int temp = table[left];
		table[left] = table[right];
		table[right] = temp;
	}
}

File: src/main/java/main/sorting_kotlin/impl/BubbleSort.kt
package main.sorting_kotlin.impl

import main.sorting_kotlin.Sort

class BubbleSort : Sort<IntArray> {

    override fun getSorted(table: IntArray): IntArray {
        for (i in table) {
            var swapped: Boolean = false
            for (index in 0 until table.size - 1) {
                if (table[index] > table[index + 1]) {
                    swap(table, index)
                    swapped = true
                }
            }
            if (!swapped) break
        }
        return table
    }

    private fun swap(table: IntArray, index: Int) {
        val temp: Int = table[index]
        table[index] = table[index + 1]
        table[index + 1] = temp
    }
}
